src/esperanza/finalizationstate.cpp

```diff
@@ -1123,4 +1112,22 @@ bool FinalizationState::IsFinalizerVoting(const Validator &finalizer) const {
   return IsInDynasty(finalizer, m_current_dynasty) || IsInDynasty(finalizer, m_current_dynasty - 1);
 }
 
+uint32_t FinalizationState::CalculateWithdrawEpoch(const Validator &finalizer) const {
+
+  // m_end_dynasty is not set
+  if (finalizer.m_end_dynasty == MAX_END_DYNASTY) {
+    return MAX_END_DYNASTY;
```

Comment:
I'm confused, the function is called `CalculateWithdrawEpoch` and we check and return a dynasty value?

Reply:
@Gnappuraz initially it was `std::numeric_limits<uint32_t>::max()`
but in this discussion https://github.com/dtr-org/unit-e/pull/1009#discussion_r277979972 it was proposed to change to `MAX_END_DYNASTY` but thinking about it again, I think it wasn't a good idea and I should revert it back to `std::numeric_limits<uint32_t>::max()`.

Will do this.

Reply:
changed